SampleARApp/SampleARApp/Examples/PlaneDetectionExample.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  PlaneDetectionExample.swift
+//  SampleARApp
+//
+//  Created by Tony Ngo on 06.09.2022.
+//
+
+import Foundation
+import ARKit
+
+final class PlaneDetectionExample: ARViewController {
+    private var isDetecting: Bool = false
+
+    private lazy var detectButton: UIButton = {
+        let button = UIButton()
+        button.layer.cornerRadius = 50
+        button.clipsToBounds = true
+        button.backgroundColor = .red.withAlphaComponent(0.7)
+        return button
+    }()
+
+    override func viewDidLoad() {
+        setupPlaneDetection()
+        super.viewDidLoad()
+        setupDetectButton()
+    }
+}
+
+// MARK: - ARSCNViewDelegate
+
+extension PlaneDetectionExample: ARSCNViewDelegate {
+    func renderer(_ renderer: SCNSceneRenderer, didAdd node: SCNNode, for anchor: ARAnchor) {
```

Comment:
what are the yellow dots I see on the camera?

Reply:
Those are called feature points. It's a point cloud showing results of the scene analysis. These data are used to help the plane detection. 

We show them by passing the option into `debugOptions` on the scene.

```swift
sceneView.debugOptions = [.showFeaturePoints]
```

Reply:
thank you, just keep that in mind :-D